example-model/src/main/java/uk/gov/gchq/palisade/example/configurator/ExampleConfigurator.java

```diff
@@ -172,40 +176,24 @@ private void addSerialiser() {
     }
 
     private FileResource createFileResource(final String id) {
-        String path = id.substring(0, id.lastIndexOf("/") + 1);
         FileResource file = new FileResource().id(id).serialisedFormat("avro").type("employee");
-        file.setParent(createParentResource(path));
+        Path path = Path.of(id).toAbsolutePath();
+        resolveParents(path.getParent(), file);
 
         return file;
     }
 
-    private DirectoryResource createParentResource(final String path) {
-        String str = path;
-        List<DirectoryResource> resourceList = new ArrayList<>();
-        List<String> pathList = new ArrayList<>();
-
-        do {
-            pathList.add(str);
-            str = str.substring(0, str.lastIndexOf("/"));
-        } while (!str.endsWith("//"));
-
-        for (String s : pathList) {
-            DirectoryResource parentResource = addParentResource(s);
-            if (!resourceList.isEmpty()) {
-                resourceList.get(resourceList.size() - 1).setParent(parentResource);
-            }
-            resourceList.add(parentResource);
-        }
-        resourceList.get(resourceList.size() - 1).setParent(createSystemResource(str));
-
-        return resourceList.get(0);
-    }
-
-    private DirectoryResource addParentResource(final String path) {
-        return new DirectoryResource().id(path);
+    private void resolveParents(final Path path, final ChildResource childResource) {
```

Comment:
I would have expected to see some unit tests to prove this method works as expected, especially around edge cases.

Reply:
Since this will be compiled before running, any given test will be testing on the same environment it will run on.
The only real requirement is that there is SystemResource / at least one DirectoryResource / FileResource